src/Orleans.Streaming.Redis/Storage/RedisDataManager.cs

```diff
@@ -158,61 +199,52 @@ private void CheckAlertSlowAccess(DateTimeOffset startOperation, [CallerMemberNa
         private void OnChannelReceivedData(RedisChannel channel, RedisValue value)
         {
             _queue.Enqueue(value);
+            _logger.Debug("{Count}", _queue.Count);
+
+            // Dequeue until we're below our queue cache limit
+            while (_queue.Count > _options.QueueCacheSize && _queue.TryDequeue(out var _)) { }
```

Comment:
Do we want to dequeue until it is below the cache size? The impression I got from the config was that it was more of a maximum queue size?

Reply:
I'm fine with this - I probably misunderstood since I now am thinking this would dequeue to just under the maximum which is likely fine since we shouldn't be encountering it anyways.